e2e/cypress/integration/emoji/sorted_emojis_spec.js

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('Filtered emojis are sorted by recency, then begins with, then contains', () => {
+    before(() => {
+        cy.apiLogin('user-1');
+        cy.visit('/');
+        cy.clearLocalStorage(/recent_emojis/);
+    });
+
+    it('should order emoji by recency', async () => {
+        // #Post a dog emoji
+        cy.postMessage(':dog:');
+
+        // #Post a cat emoji
+        cy.postMessage(':cat:');
+
+        // #Open emoji picker
+        cy.get('#emojiPickerButton').click();
+
+        // #Assert first recently used emoji has the data-test-id value of 'cat' which was the last one we sent
+        cy.get("[data-testid='emoji__item']").first().children.should('have.data-testid', 'cat');
```

Comment:
We have plugin to easily query this `data-testid`, something like
```suggestion
        cy.queryAllByTestId('emoji__item').first().children.should('have.data-testid', 'cat');
```

Reply:
done